File: OpenEdXMobile/src/main/java/org/edx/mobile/module/registration/view/RegistrationPasswordView.java
package org.edx.mobile.module.registration.view;

import android.text.InputType;
import android.view.View;

import org.edx.mobile.module.registration.model.RegistrationFormField;

/**
 * Created by rohan on 2/11/15.
 */
class RegistrationPasswordView extends RegistrationEditTextView {

    public RegistrationPasswordView(RegistrationFormField field, View view) {
        super(field, view);
        mEditText.setInputType(InputType.TYPE_CLASS_TEXT
                | InputType.TYPE_TEXT_VARIATION_PASSWORD
                | InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS);
    }

}

File: OpenEdXMobile/src/main/java/org/edx/mobile/module/registration/view/RegistrationPasswordView.kt
package org.edx.mobile.module.registration.view

import android.text.InputType
import android.view.View
import com.google.android.material.textfield.TextInputLayout
import org.edx.mobile.module.registration.model.RegistrationFormField

internal class RegistrationPasswordView(
    field: RegistrationFormField,
    view: View,
) : RegistrationEditTextView(field, view) {
    init {
        mBinding.editText.inputType = (InputType.TYPE_CLASS_TEXT
                or InputType.TYPE_TEXT_VARIATION_PASSWORD
                or InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS)
        mBinding.inputLayout.endIconMode = TextInputLayout.END_ICON_PASSWORD_TOGGLE
    }
}
